lib/ansible/modules/network/exos/exos_vlan.py

```diff
@@ -1,274 +1,285 @@
-    """
-    element_spec = dict(
-        vlan_id=dict(type='int'),
-        name=dict(),
-        interfaces=dict(type='list'),
-        delay=dict(default=10, type='int'),
-        state=dict(default='present',
-                   choices=['present', 'absent'])
-    )
-
-    aggregate_spec = deepcopy(element_spec)
-    aggregate_spec['vlan_id'] = dict(required=True)
-
-    # remove default in aggregate spec, to handle common arguments
-    remove_default_spec(aggregate_spec)
-
-    argument_spec = dict(
-        aggregate=dict(type='list', elements='dict', options=aggregate_spec),
-        purge=dict(default=False, type='bool')
-    )
-
-    argument_spec.update(element_spec)
-
-    required_one_of = [['vlan_id', 'aggregate']]
-    mutually_exclusive = [['vlan_id', 'aggregate']]
-
-    module = AnsibleModule(argument_spec=argument_spec,
-                           required_one_of=required_one_of,
-                           mutually_exclusive=mutually_exclusive,
-                           supports_check_mode=True)
-    
-    warnings = list()
-    result = {'changed': False}
-    if warnings:
-        result['warnings'] = warnings
-
-    new_config_list = map_params_to_list(module)
-    old_config_list = map_config_to_list(module)
-    requests = map_diff_to_requests(module, old_config_list, new_config_list)
-    result['requests'] = requests
-
-    if requests:
-        if not module.check_mode:
-            change_configuration(module, requests)
-        result['changed'] = True
-
-    if result['changed']:
-        check_declarative_intent_params(old_config_list, module)
-
-    module.exit_json(**result)
-
-
-if __name__ == '__main__':
+#!/usr/bin/python
+#
+# (c) 2018 Extreme Networks Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: exos_vlan
+version_added: "2.7"
+author: "Ruturaj Vyawahare (@ruturajvy)"
+short_description: Manage VLANs on Extreme Networks EXOS network devices
+description:
+  - This module provides declarative management of VLANs
+    on Extreme XOS network devices
+notes:
+  - Tested against EXOS 30.2.0.12
```

Comment:
Please test the code against the EXOS GA Release of 30.1.1.4.xos and update this field.